### ISSUE TITLE
[Bug]: sp-button doesn't support `aria-label`

### ISSUE THREAD

**Issue by WUY97** [adobe/spectrum-web-components#5658]
### Code of conduct

- [x] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-button

### Library version

1.1.1

### Expected behavior

sp-button should be able to customize aria-label attribute

### Actual behavior

for a `icon-only` `sp-button` voice over can only read `button` and cannot customize aria-label

### How can we reproduce this issue?

1. Go to https://opensource.adobe.com/spectrum-web-components/storybook/index.html?path=/story/button-black-fill--with-icon-only
2. With VoiceOver on mac on, click on the icon-only sp-button
3. Voice over reads `button` and no aria-label was exposed to clients for customization

### Sample code or abstract reproduction which illustrates the problem

```
<sp-button
    icon-only
    aria-label="lorem ipsum" // This attribute is unavailable
>
</sp-button>
```

### Severity

SEV 3

### Would you like to track this issue in Jira?

- [x] Yes, please tell me the ticket number!

**Comment by najikahalsema** [comment]
SWC-1333
